flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -307,6 +307,12 @@ private void notifyCreditAvailable() throws IOException {
         partitionRequestClient.notifyCreditAvailable(this);
     }
 
+    private void notifyNewBufferSize() throws IOException {
+        checkPartitionRequestQueueInitialized();
+
+        partitionRequestClient.notifyNewBufferSize(this);
```

Comment:
Passing the `currentBufferSize` right away instead of using the `AtomicLong` is a good idea but unfortunately, it is not possible to get rid of `this` from here because we create `ClientOutboundMessage` which requires the channel. But we indeed can pass the two arguments.